.github/workflows/build_and_test.yml

```diff
@@ -61,74 +61,100 @@ jobs:
       if: steps.filter.outputs.tket == 'true' && steps.test_package_exists.outputs.tket_package_exists == 'true'
       run: exit 1
 
-  linux:
-    name: Build and test (Linux)
-    runs-on: ubuntu-22.04
+  build_test:
+    name: Build and test
     needs: check_changes
+    strategy:
+      matrix:
+        os: ['ubuntu-22.04', 'macos-11', 'windows-2019']
```

Comment:
Yes, the difficulty is that this would require either rebuilding and testing the C++ three times, which seems wasteful, or splitting off the C++ job and making new jobs for the Python part, picking up the C++ conan package. That should be possible: I will have a think next about how to do that!